presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -86,6 +88,8 @@ public QueryInfo(
             @JsonProperty("updateType") String updateType,
             @JsonProperty("outputStage") Optional<StageInfo> outputStage,
             @JsonProperty("failureInfo") FailureInfo failureInfo,
+            @JsonProperty("failedTaskId") Optional<String> failedTaskId,
+            @JsonProperty("failedHost") Optional<String> failedHost,
```

Comment:
Since these two fields are `Optional`， we need to use `requireNonNull` to make sure they are not null. Take a look at `outputStage` above.